File: 3931-process-string-with-special-operations-i/process-string-with-special-operations-i.java
class Solution {
    public String processStr(String s) {
        StringBuilder result = new StringBuilder();
        for (char ch : s.toCharArray()) {
            if (ch >= 'a' && ch <= 'z') {
                result.append(ch);
            } else if (ch == '*') {
                if (result.length() >= 1) {
                    result.deleteCharAt(result.length() - 1);
                }
            } else if (ch == '#') {
                if (result.length() >= 1) {
                    result.append(result);
                }

            } else {
                if (result.length() >= 1) {
                    result.reverse();
                }
            }
        }
        return result.toString();
    }
}

File: arrays-and-strings/3931-process-string-with-special-operations-i/process-string-with-special-operations-i.cpp
class Solution {
public:
    string processStr(string s) {
        string result = "";
        for (char ch : s) {
            if(ch>='a' && ch<='z'){
                result+=ch;
            }
            else if(ch == '*'){
                if(result.size()>=1)
                result.pop_back();
            }
            else if(ch=='#'){
                if(result.size()>=1)
                result+=result;
            }
            else{
                if(result.size()>=1)
                reverse(result.begin(),result.end());
            }
        }
        return result;
    }
};